src/graphic/Fast3D/gfx_pc.cpp

```diff
@@ -1826,6 +1830,13 @@ static void gfx_sp_movemem_f3d(uint8_t index, uint8_t offset, const void* data)
         case G_MV_VIEWPORT:
             gfx_calc_and_set_viewport((const Vp_t*)data);
             break;
+// This is a pain but it makes the define easier to read
```

Comment:
There should not be any gbi specific ifdefs or checks inside of handlers. We should split this into two functions, one for f3dex2 and one for f3dex.